### PR TITLE
Adding anonymizeSnippets in reports config

### PR DESCRIPTION
**Pull request by dristysrivastava**
**Changes:**
- Added `anonymizedSnippets` in `reports' section in config
    - If `anonymizedSnippets` will be there in both `reprt` and `classifier`. It will give following error:

Screenshot 2024-09-24 at 12 40 57

    - If `anonymizedSnippets` will be there in `classifier`, following deprecation message will be shown:

Screenshot 2024-09-24 at 12 42 02

- Updated config custom validation with pydanticV2 model validation

- When pebblo server exists due to validation failure, now it will exits with failure message:
    - Before (Earlier it was failing with following messages)

Screenshot 2024-09-24 at 12 45 29

    - After (Now)
Screenshot 2024-09-24 at 12 47 03

- Updated `/loader/doc` API (for both file and DB) with latest config chaneg for `anonymizeSnippets`.
- Updated Unit test cases for config validation

### CODE REVIEW

File: pebblo/app/config/models.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+
+# Port BaseModel
+class DaemonConfig(BaseSettings):
+    host: str = Field(default="localhost")
+    port: int = Field(default=8000)
+
+    @field_validator("host")
+    @classmethod
+    def host_should_be_str(cls, host: str) -> str:
```
Comment (shreyas-damle) [inline pebblo/app/config/models.py:35]: One liner doc string for validation condition would be better.

File: pebblo/app/config/models.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    @field_validator("port")
+    @classmethod
+    def check_port_validity(cls, port: str) -> int:
+        try:
+            port = int(port)
+        except ValueError:
+            raise ValueError(
+                f"Error: Invalid port value '{port}'. Port must be an integer."
+            )
+        if not (0 < port <= 65535):
```
Comment (shreyas-damle) [inline pebblo/app/config/models.py:49]: When invalid value like `abc` is passed to port, this statement will give error. This check need to be inside try-except block above.

File: pebblo/app/service/loader/loader_doc_service.py [hunk 3]
```diff
@@ -302,9 +302,19 @@ def _set_anonymize_snippets(self):
...
+                )
+            elif (
+                config_details.get("classifier", {}).get("anonymizeSnippets")
+                is not None
+            ):
+                self.anonymize_snippets = config_details.get("classifier", {}).get(
+                    "anonymizeSnippets"
+                )
+            else:
+                self.anonymize_snippets = False
```
Comment (shreyas-damle) [inline pebblo/app/service/loader/loader_doc_service.py:317]: One line above each condition statement would make it easy to read!

File: pebblo/app/service/service.py [hunk 4]
```diff
@@ -143,9 +143,19 @@ def _set_anonymize_snippets(self):
...
+                )
+            elif (
+                config_details.get("classifier", {}).get("anonymizeSnippets")
+                is not None
+            ):
+                self.anonymize_snippets = config_details.get("classifier", {}).get(
+                    "anonymizeSnippets"
+                )
+            else:
+                self.anonymize_snippets = False
```
Comment (shreyas-damle) [inline pebblo/app/service/service.py:158]: This logic is duplicate. Can we move this whole function to a common place?

File: tests/app/config/test_config_validation.py [hunk 5]
```diff
@@ -1,246 +1,210 @@
...
+    assert valid_data == Config(**config_json)
+
+
+def test_daemon_config_validate_invalid_port():
+    config_json.update({"daemon": {"host": "localhost", "port": "12345678"}})
+    with pytest.raises(Exception) as err_msg:
+        Config.parse_obj(config_json)
+    error_msg = """daemon.port
+  Value error, Error: Invalid port '12345678'. Port must be between 1 and 65535."""
+    assert error_msg in str(err_msg.value)
```
Comment (shreyas-damle) [inline tests/app/config/test_config_validation.py:32]: One UT with invalid port value as a string like `invalid_string` is worth adding here.

**Review by shreyas-damle — CHANGES_REQUESTED** [review]

File: pebblo/app/config/models.py (hunk 2 above)
Comment (dristysrivastava) [reply pebblo/app/config/models.py:49]: If we are passing port as `abc` it is coming with proper error:

```
Error while loading config details, err: 1 validation error for Config
daemon.port
  Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='abc', input_type=str]
```

Comment (dristysrivastava) [reply pebblo/app/config/models.py:49]: Removed custom validation for int type checking, as pydantic is already doing it while conversion.

File: pebblo/app/service/service.py (hunk 4 above)
Comment (dristysrivastava) [reply pebblo/app/service/service.py:158]: This code block is repeated in `/loader/doc/` API file for storage type file and DB, currently we are not combining their code.

**Review by shreyas-damle — APPROVED** [review]

**Pull request merged**